### PR TITLE
Extend the release script to wait for GitHub Actions to finish and to be usable as a guide for the whole process.

### PR DESCRIPTION
**Pull request by reivilibre**
This allows you to run `./scripts-dev/release.py full` and be walked through, step-by-step, how to do a release.

### CODE REVIEW

File: scripts-dev/release.py [hunk 1]
```diff
@@ -548,10 +636,55 @@ def announce() -> None:
...
+    click.echo("\n*** upload ***")
+    _upload()
+
+    click.echo("\nWait for all release methods to be ready.")
+    # Docker should be ready because it was done by the workflows earlier
+    # PyPI should be ready because we just ran upload().
+    click.confirm("Debs ready?", abort=True)
+
+    click.echo("\n*** merge back ***")
+    _merge_back()
```
Comment (erikjohnston) [inline scripts-dev/release.py:682]: I'd move this above waiting for the debs.

File: scripts-dev/release.py [hunk 2]
```diff
@@ -548,10 +636,55 @@ def announce() -> None:
...
+    click.echo("\nUpdate the Debian repository")
+    click.confirm("Started updating Debian repository?", abort=True)
+
+    click.echo("\n*** upload ***")
+    _upload()
+
+    click.echo("\nWait for all release methods to be ready.")
+    # Docker should be ready because it was done by the workflows earlier
+    # PyPI should be ready because we just ran upload().
+    click.confirm("Debs ready?", abort=True)
```
Comment (erikjohnston) [inline scripts-dev/release.py:679]: We should be able to poll for the debs too

File: scripts-dev/release.py [hunk 3]
```diff
@@ -548,10 +636,55 @@ def announce() -> None:
...
+    _wait_for_actions()
+
+    click.echo("\n*** publish ***")
+    _publish(gh_token)
+
+    click.echo("\nUpdate the Debian repository")
+    click.confirm("Started updating Debian repository?", abort=True)
+
+    click.echo("\n*** upload ***")
+    _upload()
```
Comment (erikjohnston) [inline scripts-dev/release.py:674]: I'd upload before the prompt? Once we've published we may as well upload the assets immediately.

**Review by erikjohnston — COMMENTED** [review]
Woo automation.

File: scripts-dev/release.py (hunk 2 above)
Comment (reivilibre) [reply scripts-dev/release.py:679]: I suppose we can hit https://packages.matrix.org/debian/pool/main/m/matrix-synapse-py3/matrix-synapse-py3_1.66.0%2Bbuster1.dsc etc if that's what you mean?

File: scripts-dev/release.py (hunk 3 above)
Comment (reivilibre) [reply scripts-dev/release.py:674]: That's fair, I was just doing things in the same order as the instructions on the wiki

File: scripts-dev/release.py [hunk 4]
```diff
@@ -501,19 +501,23 @@ def _merge_into(repo: Repo, source: str, target: str) -> None:
 
 
 @cli.command()
-def wait_for_actions() -> None:
-    _wait_for_actions()
+@click.option("--gh-token", envvar=["GH_TOKEN", "GITHUB_TOKEN"], required=True)
```
Comment (erikjohnston) [inline scripts-dev/release.py:504]: It's not required?

Comment (reivilibre) [reply scripts-dev/release.py]: Oops.

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**